packages/storefront-api-client/src/storefront-api-client.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  createGraphQLClient,
+  CustomFetchApi,
+  RequestParams as GQLClientRequestParams,
+  getCurrentSupportedApiVersions,
+  validateDomainAndGetStoreUrl,
+  validateApiVersion,
+  ApiClientLogger,
+  ApiClientRequestParams,
+  ApiClientRequestOptions,
+} from "@shopify/graphql-client";
+
+import {
+  StorefrontApiClient,
+  StorefrontApiClientConfig,
+  StorefrontApiClientLogContentTypes,
+} from "./types";
+import {
+  DEFAULT_SDK_VARIANT,
+  DEFAULT_CLIENT_VERSION,
+  SDK_VARIANT_HEADER,
+  SDK_VARIANT_SOURCE_HEADER,
+  SDK_VERSION_HEADER,
+  DEFAULT_CONTENT_TYPE,
+  PUBLIC_ACCESS_TOKEN_HEADER,
+  PRIVATE_ACCESS_TOKEN_HEADER,
+  CLIENT,
+} from "./constants";
+import {
+  validateRequiredAccessTokens,
+  validatePrivateAccessTokenUsage,
+} from "./validations";
+
+export function createStorefrontApiClient({
+  storeDomain,
+  apiVersion,
+  publicAccessToken,
+  privateAccessToken,
+  clientName,
+  retries = 0,
+  customFetchApi: clientFetchApi,
+  logger,
+}: {
+  storeDomain: string;
+  apiVersion: string;
+  clientName?: string;
+  retries?: number;
+  customFetchApi?: CustomFetchApi;
+  logger?: ApiClientLogger<StorefrontApiClientLogContentTypes>;
+} & (
+  | {
+      publicAccessToken?: never;
+      privateAccessToken: string;
+    }
+  | {
+      publicAccessToken: string;
+      privateAccessToken?: never;
+    }
+)): StorefrontApiClient {
```

Comment:
I kept it like this because in general, by writing this out, it allows the editor code hints to show the type object instead of just the name of the interface/type. I don't feel strongly about it so I'm good to go either way - do you feel strongly about it?

Reply:
I feel like this is essentially duplicating the actual type we have, so we'd need to change two places if this changed - plus, the other type doesn't have the token logic in it, which is already a small bit of discrepancy between them. I'd personally prefer for this to be the type, and we rely on autocomplete more so than type hints for calling this function.